### PR TITLE
Display summary data per deployment

### LINKED ISSUE #696: Display summary data per deployment
Current implementation of discovery engine generates summary data per pod. If there are multiple pods for a single deployment/replicasets then the summary will be lengthy. To reduce the noise and show a clean summary data,  discovery engine should generate the data per deployment. This will make sure that all the data from pods of deployment/statefulset will be merged under one table and shown.

### PR DESCRIPTION
**Pull request by vishnusomank**
- Modified the observability to include Summary data per deployment
- Added deployment name to struct PodInfo
- Modified codebase to get details of replicasets and statefulset along with deployments
- Modified codebase to include deployment name as part of PodInfo

Fixes: https://github.com/accuknox/discovery-engine/issues/696

### Related PR's
1. [kubearmor-client #299](https://github.com/kubearmor/kubearmor-client/pull/299)

### CODE REVIEW

File: src/cluster/k8sClientHandler.go
Comment (seswarrajan) [inline src/cluster/k8sClientHandler.go:1]: Kindly handle lint errors.

**Comment by Vyom-Yadav** [comment]
@vishnusomank Can you share the output as well?

Also, the summary data may vary per pod if someone exec'd into that particular pod, is that something we should consider?

**Comment by vishnusomank** [comment]
> @vishnusomank Can you share the output as well?
> 
> Also, the summary data may vary per pod if someone exec'd into that particular pod, is that something we should consider?

@Vyom-Yadav there is no change in the codebase we are only sending the data arranged by deployment name instead of pod name. So internally the storage and function calls are almost the same.

You can refer this PR for output https://github.com/kubearmor/kubearmor-client/pull/299

File: src/observability/summarizer.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func convertSysLogToSysSummaryMap(syslogs []*pb.Alert) {
 		sysSummary.Deployment = ""
 
 		for _, d := range deployments {
-			if d.Labels == syslog.Labels && d.Namespace == syslog.NamespaceName {
+			if strings.Contains(syslog.Labels, d.Labels) && d.Namespace == syslog.NamespaceName {
```
Comment (seswarrajan) [inline src/observability/summarizer.go:130]: I need clarification in this logic @vishnusomank 
Why are we using strings.Contains instead of == ?

Comment (vishnusomank) [reply src/observability/summarizer.go:130]: @seswarrajan  The `syslog.Labels` sometimes contains multiple labels as a single string and when we use == to compare it with `d.Labels` it fails. So to reduce the failure rate I added the `strings.Contains` condition.

eg: 
syslog.Labels 
 ```
sidecar.istio.io/inject=false,app=istiod,install.operator.istio.io/owning-resource=unknown,istio=pilot istio.io/rev=default,operator.istio.io/component=Pilot
```
d.Labels
```
app=istiod
```

In this scenario, the pod belongs to the deployment `istiod` but if we use == it'll not be able to register.

File: src/cluster/k8sClientHandler.go [hunk 2]
```diff
@@ -458,6 +457,92 @@ func GetDeploymentsFromK8sClient() []types.Deployment {
...
+	for _, rs := range replicasets.Items {
+		if rs.OwnerReferences == nil {
+			if rs.Namespace == "kube-system" {
+				continue
+			}
+
+			var label string
+
+			for k, v := range rs.Spec.Selector.MatchLabels {
+				label = k + "=" + v
```
Comment (Ankurk99) [inline src/cluster/k8sClientHandler.go]: Do we need to handle the condition when label is empty?

Comment (Ankurk99) [reply src/cluster/k8sClientHandler.go]: With multiple MatchLabels, the label variable will be overwritten (inside for loop) and only the last key-value pair in MatchLabels will be stored in it?

File: src/cluster/k8sClientHandler.go [hunk 3]
```diff
@@ -458,6 +457,92 @@ func GetDeploymentsFromK8sClient() []types.Deployment {
...
+	for _, sts := range statefulset.Items {
+		if sts.OwnerReferences == nil {
+			if sts.Namespace == "kube-system" {
+				continue
+			}
+
+			var label string
+
+			for k, v := range sts.Spec.Selector.MatchLabels {
+				label = k + "=" + v
```
Comment (Ankurk99) [inline src/cluster/k8sClientHandler.go]: Same as above. How are we using the multiple `label` values?

**Review by Ankurk99 — CHANGES_REQUESTED** [review]
Please find comments inline

File: src/cluster/k8sClientHandler.go [hunk 4]
```diff
@@ -446,17 +445,112 @@ func GetDeploymentsFromK8sClient() []types.Deployment {
...
 
-		var label string
+		if d.Spec.Selector.MatchLabels != nil {
+			var labels []string
 
-		for k, v := range d.Spec.Selector.MatchLabels {
-			label = k + "=" + v
+			for k, v := range d.Spec.Selector.MatchLabels {
+				label := k + "=" + v
+				labels = append(labels, label)
```
Comment (Ankurk99) [inline src/cluster/k8sClientHandler.go]: what about using `labels = append(labels, k+"="+v)` ?

File: src/cluster/k8sClientHandler.go [hunk 5]
```diff
@@ -446,17 +445,112 @@ func GetDeploymentsFromK8sClient() []types.Deployment {
...
+// ================= //
+// == ReplicaSet == //
+// ================= //
+
+func GetReplicaSetsFromK8sClient() []types.Deployment {
+	results := []types.Deployment{}
+
+	client := ConnectK8sClient()
+	if client == nil {
+		return results
```
Comment (Ankurk99) [inline src/cluster/k8sClientHandler.go:489]: Should we log error or handle the case when `results` is empty?

**Review by seswarrajan — APPROVED** [review]
LGTM

**Review by Ankurk99 — APPROVED** [review]
Please handle the go-lint warnings, rest LGTM

**Pull request merged**